serving/ingress/pkg/reconciler/ingress/resources/route.go

```diff
@@ -154,6 +167,23 @@ func makeRoute(ci *networkingv1alpha1.Ingress, host string, rule networkingv1alp
 	return route, nil
 }
 
+// isInternalEncryptionEnabled determines whether internal-encryption is enabled or not.
+// In general, we can determine it by the value internal-encryption in config-network, however the serverless ingress does not
+// watch the ConfigMap. Therefore we determine it by ServiceHTTPSPort(443) port for the backend in Kingress.
+func isInternalEncryptionEnabled(rule networkingv1alpha1.IngressRule) bool {
```

Comment:
Can we add a watch for the configmap or is this robust enough?
A more general question. In the future what stops us from moving the ingress stuff to the knative operator besides the overhead of the resources? 
As a side note information such as if the internal encryption is on are provided for free at the Serving CR level.
I am asking because I am working on the vision of having one operator for all stuff. Wdyth?

Reply:
I think it is robust enough. Actually adding a watch for the configmap has two problems:
1. We need to pull informer/listner code.
2. It would cause a conflictions between Kingress and OpenShift Ingress in just a few (milli)seconds. (Because they are watching the configmap via different controller.)
So, watching the ingress would be better.

Reply:
I think it is better to answer in another Slack or somewhere? 
If we started the thread here, we cannot merge this :sweat_smile: